tools/topic_tools/src/throttle.cpp

```diff
@@ -57,6 +57,7 @@ bool g_use_messages;
 ros::Time g_last_time;
 bool g_use_wallclock;
 bool g_lazy;
+int g_latch = -1;
```

Comment:
Please use a boolean parameter for this and rename it to `force_latch`.

Reply:
I actually need to distinguish between three values:
- Make the publisher `latching` property the same as the subscriber
- Make the publisher latched, regardless of the subscriber `latching` property
- Make the publisher non-latched, regardless of the subscriber `latching` property

So I can't use a `bool force_latch` only. If you prefer, I can have two variables:
- `bool force_latch;` : Whether to force the user given `latch` value, or use the subscriber one
- `bool latch` : The desired `latching` value when forcing it

This means I'd have to read two ROS params, which is fine. They'd default to `false`, to keep the current behaviour by default.

Regarding the name, I was trying to follow the `g_` pattern used for the other global variables, but I'm happy to get rid of the global variables and create a class. But I'd do that in another PR. Thoughts? :smiley:

Reply:
> So I can't use a `bool force_latch` only.

Why not? if the parameter is not set the behavior stays as it was before - mimic the property of the subscriber.

If the parameter is set it overrides the property with the provided boolean.

Reply:
You can do that for the ROS parameter:
1. If the ROS parameters isn't set: previous behaviour, i.e. output topic latching == input topic latching
2. If the ROS parameter is set to `false`, output topic latching == false
3. If the ROS parameter is set to `true`, output topic latching == true

So we can use a single `force_latch` ROS parameter.

That's ok, but, as I said, I need a copule of `bool` variables to handle those three states.

I've done that. I think we weren't talking about the same thing.

If tested things out for the 6 possible combinations:

| Force Latch? | Input Topic Latch? | Output Topic Latch? |
|--------------|--------------------|---------------------|
| True         | True               | True                |
| True         | False              | True                |
| False        | True               | False               |
| False        | False              | False               |
| Not set      | True               | True                |
| Not set      | False              | False               |

I used:
1. `rostopic pub -r 1 /test std_msgs/Float32 1` for a non-latched input topic
2. `rostopic pub -l -r 1 /test std_msgs/Float32 1` for a non-latched input topic

And:
1. `rosrun topic_tools throttle messages /test 0.1 _force_latch:=true` to force the output topic to be latched
2. `rosrun topic_tools throttle messages /test 0.1 _force_latch:=false` to force the output topic to be non-latched
3. `rosrun topic_tools throttle messages /test 0.1` to set the output topic latching the same as the input topic one (previous bahaviour)